Makefile

```diff
@@ -13,36 +13,48 @@
 # limitations under the License.
 
 # Old-skool build tools.
-# Simple makefile to build kind quickly and reproducibly in a container..
-# Only requires docker on the host
+# Simple makefile to build kind quickly and reproducibly
 #
 # Common uses:
 # - installing kind: `make install INSTALL_DIR=$HOME/go/bin`
 # - building: `make build`
 # - cleaning up and starting over: `make clean`
-
-# get the repo root and output path, go_container.sh respects these
+#
+################################################################################
+# ========================== Capture Environment ===============================
+# get the repo root and output path
 REPO_ROOT:=${CURDIR}
 OUT_DIR=$(REPO_ROOT)/bin
+# record the source commit in the binary, overridable
+COMMIT?=$(shell git rev-parse HEAD 2>/dev/null)
+################################################################################
+# ========================= Setup Go With Gimme ================================
+# go version to use for build etc.
+# setup correct go version with gimme
+PATH:=$(shell . hack/build/setup-go.sh && echo "$${PATH}")
```

Comment:
- GOROOT is not required since go.19, go will look next to the go binary on it's own
- I don't think GOOS / GOARCH are relevant here
- PATH is how we inject the go version we want

This part is also not _striclty_ necessary after a later commit, where scripts try to ensure this themselves

Reply:
@amwat pointed out that gvm sets this (and probably other tools, including gimme when used natively), and when set to a different value in the host environment that causes us to potentially use the gimme go binary against some other goroot. we mitigate that below now by clearing GOROOT so it will be auto-detected instead.